static/js/chart/draw.ts

```diff
@@ -90,7 +90,17 @@ function appendLegendElem(
     .attr("style", (d) => `background: ${color(d.label)}`)
     .append("a")
     .text((d) => d.label)
-    .attr("href", (d) => d.link || null);
+    .attr("href", (d) => d.link || null)
+    // Triggered when stat var legend is clicked and send data to google analytics.
+    .on(
+      "click",
+      () =>
+        window &&
```

Comment:
window is a global, so we can safely skip this check. curious if you saw documentation / examples of when it might not be the case?

Reply:
When I ran it locally, sometimes the whole page cannot be loaded and the error message is window.gtag is not a function. I took the answers on https://stackoverflow.com/questions/62791299/typeerror-window-gtag-is-not-a-function as a reference.

Reply:
i'm surprised and not sure if i buy those answers. as you can see in our code, we never check for the existence of window, and window.gtag is defined at the head of the template before any other script is run. when exactly did you see that error happen?

Reply:
It occurred randomly...Window should already exist and we can skip checking that if we never check for the existence of window. But we may still need to check window.gtag because this script is async and I did encounter some errors...
https://github.com/datacommonsorg/website/blob/master/server/templates/base.html#L39

Reply:
Done